debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlStreamingChangeEventSource.java

```diff
@@ -639,10 +639,16 @@ protected void handleUpdateTableMetadata(MySqlPartition partition, MySqlOffsetCo
      * don't know, either ignore that event or raise a warning or error as per the
      * {@link MySqlConnectorConfig#INCONSISTENT_SCHEMA_HANDLING_MODE} configuration.
      */
-    private void informAboutUnknownTableIfRequired(MySqlPartition partition, MySqlOffsetContext offsetContext, Event event, TableId tableId, String typeToLog)
+    private void informAboutUnknownTableIfRequired(MySqlPartition partition, MySqlOffsetContext offsetContext, Event event, TableId tableId, String typeToLog,
+                                                   Operation operation)
             throws InterruptedException {
         if (tableId != null && connectorConfig.getTableFilters().dataCollectionFilter().isIncluded(tableId)) {
-            metrics.onErroneousEvent("source = " + tableId + ", event " + event);
+            if (operation != null) {
```

Comment:
Ok, seeing this, I think we should always call the method _with_ the operation parameter here:

    // operation may be null
    metrics.onErroneousEvent("source = " + tableId + ", event " + event, operation);

What I was trying to get at when suggesting to have two overloads of that methods was to simplify the case when _no_ operation is present at all in a given context: in this case the overload without the additional parameter can be invoked, without passing `null`.

Reply:
Got it